source/server/server.cc

```diff
@@ -134,10 +135,10 @@ void InstanceImpl::flushStats() {
   stat_flush_timer_->enableTimer(config_->statsFlushInterval());
 }
 
-int InstanceImpl::getListenSocketFd(uint32_t port) {
+int InstanceImpl::getListenSocketFd(const std::string& address) {
+  Network::Address::InstancePtr addr = Network::Utility::resolveUrl(address);
```

Comment:
No point in resolving this, just do the string compare directly. Either it matches or it doesn't.